packages/wrangler/src/api/startDevWorker/types.ts

```diff
@@ -129,7 +129,7 @@ export interface StartDevWorkerInput {
 		/** Whether the worker runs on the edge or locally. */
 		remote?: boolean;
 		/** Cloudflare Account credentials. Can be provided upfront or as a function which will be called only when required. */
-		auth?: AsyncHook<CfAccount>;
+		auth?: AsyncHook<CfAccount, [Pick<Config, "account_id">]>; // provide config.account_id as a hook param
```

Comment:
only the ConfigController has access to the wrangler.toml config so enable it as a hook param for `StartDevWorkerInput.dev.auth`